### PR TITLE
remove scaler from ckpt if in bf16 mode

### PR DESCRIPTION
**Pull request by Adel-Moumen**
## What does this PR do?

This PR remove the `gradscaler` from the `checkpoints` in `bf16` as indeed, in `bf16` there's no need of `gradscaler` as it is set to `enabled=False`. In the past, we got the following warning when using `bf16` and checkpointing:
```
speechbrain.utils.checkpoints - Saving
an empty state_dict for <torch.cuda.amp.grad_scaler.GradScaler object at 0x7effb93bb490> in results/whole-lattice-rescoring/24/save/CKPT+2024-01-22+12-47-56+00/scaler.ckpt.
speechbrain.utils.checkpoints - Saving an empty state_dict for <torch.cuda.amp.grad_scaler.GradScaler object at 0x7effb93bb490> in results/whole-lattice-rescoring/24/save/CKPT+2024-01-22+13-13-13+00/scaler.ckpt.
```

<details>
  <summary><b>Before submitting</b></summary>

- [x] Did you read the [contributor guideline](https://speechbrain.readthedocs.io/en/latest/contributing.html)?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing [tests](https://github.com/speechbrain/speechbrain/tree/develop/tests) pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Does your code adhere to project-specific code style and conventions?

</details>

## PR review

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
- [x] Confirm that the changes adhere to compatibility requirements (e.g., Python version, platform)
- [x] Review the self-review checklist to ensure the code is ready for review

</details>

### CODE REVIEW

**Review by TParcollet — APPROVED** [review]
ok.
